scripts/logging.sh

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env bash
+#
+# Helper functions for console (stdout/stderr) logging.
+#
+
+COLOR_GREEN="\033[0;32m"
+COLOR_YELLOW="\033[0;33m"
+COLOR_RED="\033[0;31m"
+COLOR_RESET="\033[0m"
+
+# Helper function, not to use directly.
+# Emits the current timestamp to stdout to be used as header in logging traces.
+# If `date` command fails, it will print the error output instead and will
+# exit with status code 1.
+timestamp() {
+    local tms
+	tms="$(date +%Y%m%d-%H:%M:%S 2>&1)" || {
+		printf >&2 "%s\n" "Failed to run date command: $tms"
+		exit 1
+	}
+	printf %s "$tms"
+}
+
+# Info logger: emits arguments to stdout with date-time and "INFO" as header.
+# Example:
+#  20230531-02:41:52 INFO : test for tracing with log_info
+log_info() {
+	local tms
+	tms="$(timestamp)" || exit $?
+	printf "${COLOR_GREEN}%s INFO : ${COLOR_RESET}%s\n" "$tms" "$*"
+}
+
+# Warning logger: emits arguments to stderr with date-time and "WARN" as header.
+# Example:
+#  20230531-02:41:52 WARN : test for tracing with log_warn
+log_warn() {
+	local tms
+	tms="$(timestamp)" || exit $?
+	printf >&2 "${COLOR_YELLOW}%s WARN : ${COLOR_RESET}%s\n" "$tms" "$*"
+}
+
+# Error logger: emits arguments to stderr with date-time and "ERROR" as header.
+#  20230531-02:41:52 ERROR: test for tracing with log_error
+log_error() {
+	local tms
+	tms="$(timestamp)" || exit $?
+	printf >&2 "${COLOR_RED}%s ERROR: ${COLOR_RESET}%s\n" "$tms" "$*"
```

Comment:
```suggestion
	printf >&2 "${COLOR_RED}%s ERROR : ${COLOR_RESET}%s\n" "$tms" "$*"
```

Reply:
Did you mean ERROR without the space? I'd remove `:` at all.

By the way, it is a good idea to add the log functions in separate commit.

Reply:
> For vertical alignment?

Yes.

> By the way, it is a good idea to add the log functions in a separate commit.

You're right, but it's too late. I won't rebuild this PR's history anymore.

As the rest is just cosmetics, totally unrelated with the original scope of this PR, I prefer either waiting for other people's comments on it or just merge it as it is, so you can change it as you please in further PRs.